### PR TITLE
Collate packed sequences

### PR DESCRIPTION
**Pull request by alexandresablayrolles**
Support for tuples as well as single tensors

### CODE REVIEW

File: opacus/data_loader.py [hunk 1]
```diff
@@ -10,21 +10,18 @@
 def wrap_collate_with_empty(
     collate_fn: Optional[_collate_fn_t], sample_empty_shapes: Sequence
 ):
-    # TODO: does it work with packed sequences?
     def collate(batch):
         if len(batch) > 0:
             return collate_fn(batch)
         else:
-            return [torch.zeros(x) for x in sample_empty_shapes]
+            if type(sample_empty_shapes) is collections.abc.Sequence:
```
Comment (karthikprasad) [inline opacus/data_loader.py:17]: This should be isinstance() instead. 
`type(list()) is collections.abc.Sequence` will return False
`isinstance(list(), collections.abc.Sequence)` will return True

File: opacus/data_loader.py [hunk 2]
```diff
@@ -10,21 +10,18 @@
...
+                return [torch.zeros(x) for x in sample_empty_shapes]
+            else:
+                return torch.zerso(sample_empty_shapes)
```
Comment (karthikprasad) [inline opacus/data_loader.py:20]: typo: zeros

**Review by karthikprasad — CHANGES_REQUESTED** [review]

**Comment by ffuuugor** [comment]
Do we want to have a test for this, to make sure it works as expected?

**Comment by alexandresablayrolles** [comment]
Turns out the original code was doing just fine...reverted my changes but still keeping tests in case it's useful.

**Review by ffuuugor — APPROVED** [review]
Yay, more tests!
Please fix flake8 errors before merging

**Pull request merged**